crates/sdk/src/tx.rs

```diff
@@ -3081,7 +3083,7 @@ pub async fn build_shielding_transfer<N: Namada>(
         source: source.clone(),
         token: args.token.clone(),
         amount: validated_amount,
-        shielded_section_hash: Hash::zero(),
+        shielded_section_hash: TxId::from_bytes(Default::default()).into(),
```

Comment:
Is it enough to use the `TxId` as the commitment to the shielded section? The hash of the `data` section ends up in the `TxCommitments` which in turn is part of the `raw_header_hash` that gets signed by the inner tx signer(s). If this is malleable then it's still possible to tamper with the masp section. Since the hash of the masp `Transaction` is now used when computing the hash of the masp section, once the wrapper has been constructed it's safe but it is still possible for the wrapper signer to modify that section before wrapping the inner batch. Of course this is a less sever issue than tampering with the wrapper since:

- The wrapper cannot be tampered with (as stated above) so the correctness of fee payment is preserved
- We can expect in most cases the wrapper signer to be the same entity that generated the inner batch
- Tampering with the proofs/signatures could only make the transaction fail (to my understanding), so essentially it turns the tx into a no-op since it's going to be rejected. This is exactly what already happens if the wrapper signer tampers with other sections of the inner tx (like code or data): the tx either fails during execution or is rejected by vps

So overall since to me like we are safe (as long as my assumptions that tampering with proofs/signatures can only make the tx fail) but I still wanted to bring up this topic before approving the pr